pkg/ossm/smcp_install.go

```diff
@@ -55,29 +60,35 @@ func installDefaultSMCP22() {
 }
 
 func TestSMCPInstall(t *testing.T) {
-	defer installDefaultSMCP23()
+	NewTest(t).LegacyID("A1").Groups(Smoke, Full, InterOp, ARM).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		t.Cleanup(func() {
+			installDefaultSMCP(t, meshNamespace)
+		})
+		vars := vars{
+			Name:      smcp.Name,
+			Namespace: smcp.Namespace,
+		}
 
+		t.NewSubTest("smcp_test_install_2.3").Run(func(t TestHelper) {
+			t.LogStep("Crete Namespace and Install SMCP v2.3")
+			oc.CreateNamespace(t, meshNamespace)
+			oc.ApplyString(t, meshNamespace, template.Run(t, smcpV23_template, vars))
```

Comment:
I think it makes sense to create a new function `oc.ApplyTemplate()` that will run the template and then call `ApplyString()`. This way, any test that needs to apply a string that was generated from a template, can remain short and simple.

Reply:
Added
`func ApplyTemplate(t test.TestHelper, ns string, yaml string, vars interface{}) {
	t.T().Helper()
	template := template.Run(t, yaml, vars)
	if err := util.KubeApplyContents(ns, template); err != nil {
		t.Fatalf("Failed to apply manifest: %v;\nYAML: %v", err, template)
	}
}`